person_ui.rb

```diff
@@ -0,0 +1,69 @@
+class PersonUserInterface
+  attr_reader :people
+
+  def initialize
+    @people = []
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    @num = gets.chomp
+    case @num
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please enter correct number'
+    end
+  end
+
+  def list_all_people
+    if @people.count.zero?
+      puts ''
+      puts 'There are no people yet try to create one !'
+    else
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+    puts ''
+  end
+
+  private
+
+  def create_student
+    print 'Age: '
+    @age = gets.chomp
+
+    print 'Name: '
+    @name = gets.chomp
+
+    print 'Has parent_permission ? [Y/N]: '
+    @permission = gets.chomp
+
+    @permission = false if @permission == 'n'
+    @permission = true if @permission == 'y'
+
```

Comment:
- [ ] Here, the requests for **`age`** and **`name`** break 2 rules: the single-responsibility rule (should be handled by another method) and the DRY rule (it is repeated in the **create_teacher** method). Kindly extract these requests into another method and use that method instead.
- [ ] The user inputs and validation for **`@permission`** should be handled by another method. Kindly do that.

  **SUGGESTION:**

  ![elshamy-person](https://user-images.githubusercontent.com/36542986/157078364-526662f2-f083-4eea-9148-5dbdac79245e.jpg)